document_page/models/document_page.py

```diff
@@ -99,60 +117,58 @@ class DocumentPage(models.Model):
         readonly=True
     )
 
-    def _get_page_index(self, page, link=True):
+    @api.multi
+    def _get_page_index(self, link=True):
         """Return the index of a document."""
+        self.ensure_one()
         index = []
-        for subpage in page.child_ids:
-            index += ["<li>" + self._get_page_index(subpage) +
-                      "</li>"]
+        for subpage in self.child_ids:
+            index += ["<li>" + subpage._get_page_index() + "</li>"]
         r = ''
         if link:
-            r = '<a href="#id=%s">%s</a>' % (page.id, page.name)
+            r = '<a href="#id=%s">%s</a>' % (self.id, self.name)
 
         if index:
             r += "<ul>" + "".join(index) + "</ul>"
         return r
 
-    def _get_display_content(self):
-        """Return the content of a document."""
-        for page in self:
-            if page.type == "category":
-                display_content = self._get_page_index(page, link=False)
+    @api.multi
```

Comment:
Use api.depends for this compute field.

Reply:
Since it's not stored, I think it's not actually needed.

Reply:
I've just checked and indeed I've already added depends here: https://github.com/OCA/knowledge/pull/155/commits/dbeec6ef0f76f629bdf2d96c469c2fa4d25f20c7

Reply:
@zaoral where you able to look at this?

Reply:
@ivantodorovich yes thank you for apply the api.depends. LGTM